### DISCUSSION TITLE
I keep circling the same steps. Here's a log...

### DISCUSSION THREAD

**Discussion by MK00000000000000000000** [Jip-Hop/jailmaker#205] [General]
Hi, anyone have seen this issue? I keep following the steps. I create 'docker' in JLMKR and it always will not start properly.

Thanks!

`Create a new jail? [Y/n] y

USE THIS SCRIPT AT YOUR OWN RISK!
IT COMES WITHOUT WARRANTY AND IS NOT SUPPORTED BY IXSYSTEMS.

Do you wish to create a jail from a config template? [y/N] y

A text editor will open so you can provide the config template.

  1. Please copy your config
  2. Paste it into the text editor
  3. Save and close the text editor

Press Enter to open the text editor.

Enter jail name: docker

Do you want to start this jail now (when create is done)? [Y/n] n

Creating ZFS Dataset ssdev02/jailmaker/jails/docker
Using image from local cache
Unpacking the rootfs

---
You just created a Debian bookworm amd64 (20240621_05:24) container.

To enable SSH, run: apt install openssh-server
No default root or user password are set by LXC.
root@truenas[/mnt/ssdev02/jailmaker]# jlmkr list         
NAME   RUNNING STARTUP GPU_INTEL GPU_NVIDIA OS     VERSION ADDRESSES
docker False   True    False     False      debian 12      -        
root@truenas[/mnt/ssdev02/jailmaker]# jlmkr start docker 

Starting jail docker with the following command:

systemd-run --property=KillMode=mixed --property=Type=notify --property=RestartForceExitStatus=133 --property=SuccessExitStatus=133 --property=Delegate=yes --property=TasksMax=infinity --collect --setenv=SYSTEMD_NSPAWN_LOCK=0 --unit=jlmkr-docker --working-directory=./jails/docker '--description=My nspawn jail docker [created with jailmaker]' --property=ExecStartPre=/mnt/ssdev02/jailmaker/jails/docker/.ExecStartPre -- systemd-nspawn --keep-unit --quiet --boot --bind-ro=/sys/module --inaccessible=/sys/module/apparmor --machine=docker --directory=rootfs --notify-ready=yes --network-macvlan=enp4s0 --resolv-conf=bind-host '--system-call-filter=add_key keyctl bpf'

Job for jlmkr-docker.service failed.
See "systemctl status jlmkr-docker.service" and "journalctl -xeu jlmkr-docker.service" for details.

Failed to start jail docker...
In case of a config error, you may fix it with:
jlmkr.py edit docker

root@truenas[/mnt/ssdev02/jailmaker]# journalctl -xeu jlmkr-docker.service
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit jlmkr-docker.service has finished successfully.
░░ 
░░ The job identifier is 823676.
Jun 22 23:12:45 truenas systemd-nspawn[3667911]: systemd 252.22-1~deb12u1 running in system mode (+PAM +AUDIT +SELINUX +APPARMOR +IMA +SMACK +SECCOMP +GCRYPT -GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IP>
Jun 22 23:12:45 truenas systemd-nspawn[3667911]: Detected virtualization systemd-nspawn.
Jun 22 23:12:45 truenas systemd-nspawn[3667911]: Detected architecture x86-64.
Jun 22 23:12:45 truenas systemd-nspawn[3667911]: Detected first boot.
Jun 22 23:12:45 truenas systemd-nspawn[3667911]: 
Jun 22 23:12:45 truenas systemd-nspawn[3667911]: Welcome to Debian GNU/Linux 12 (bookworm)!
Jun 22 23:12:45 truenas systemd-nspawn[3667911]: 
Jun 22 23:12:45 truenas systemd-nspawn[3667911]: Initializing machine ID from container UUID.
Jun 22 23:12:45 truenas systemd-nspawn[3667911]: Failed to create control group inotify object: Too many open files
Jun 22 23:12:45 truenas systemd-nspawn[3667911]: Failed to allocate manager object: Too many open files
Jun 22 23:12:45 truenas systemd-nspawn[3667911]: [!!!!!!] Failed to allocate manager object.
Jun 22 23:12:45 truenas systemd-nspawn[3667911]: Exiting PID 1...
Jun 22 23:12:45 truenas systemd[1]: jlmkr-docker.service: Main process exited, code=exited, status=255/EXCEPTION
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ An ExecStart= process belonging to unit jlmkr-docker.service has exited.
░░ 
░░ The process' exit code is 'exited' and its exit status is 255.
Jun 22 23:12:45 truenas systemd[1]: jlmkr-docker.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The unit jlmkr-docker.service has entered the 'failed' state with result 'exit-code'.
Jun 22 23:13:15 truenas systemd[1]: Starting jlmkr-docker.service - My nspawn jail docker [created with jailmaker]...
░░ Subject: A start job for unit jlmkr-docker.service has begun execution
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit jlmkr-docker.service has begun execution.
░░ 
░░ The job identifier is 823756.
Jun 22 23:13:15 truenas .ExecStartPre[3679316]: PRE_START_HOOK
Jun 22 23:13:15 truenas systemd[1]: Started jlmkr-docker.service - My nspawn jail docker [created with jailmaker].
░░ Subject: A start job for unit jlmkr-docker.service has finished successfully
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit jlmkr-docker.service has finished successfully.
░░ 
░░ The job identifier is 823756.
Jun 22 23:13:15 truenas systemd-nspawn[3679318]: systemd 252.22-1~deb12u1 running in system mode (+PAM +AUDIT +SELINUX +APPARMOR +IMA +SMACK +SECCOMP +GCRYPT -GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IP>
Jun 22 23:13:15 truenas systemd-nspawn[3679318]: Detected virtualization systemd-nspawn.
Jun 22 23:13:15 truenas systemd-nspawn[3679318]: Detected architecture x86-64.
Jun 22 23:13:15 truenas systemd-nspawn[3679318]: Detected first boot.
Jun 22 23:13:15 truenas systemd-nspawn[3679318]: 
Jun 22 23:13:15 truenas systemd-nspawn[3679318]: Welcome to Debian GNU/Linux 12 (bookworm)!
Jun 22 23:13:15 truenas systemd-nspawn[3679318]: 
Jun 22 23:13:15 truenas systemd-nspawn[3679318]: Initializing machine ID from container UUID.
Jun 22 23:13:15 truenas systemd-nspawn[3679318]: Failed to create control group inotify object: Too many open files
Jun 22 23:13:15 truenas systemd-nspawn[3679318]: Failed to allocate manager object: Too many open files
Jun 22 23:13:15 truenas systemd-nspawn[3679318]: [!!!!!!] Failed to allocate manager object.
Jun 22 23:13:15 truenas systemd-nspawn[3679318]: Exiting PID 1...
Jun 22 23:13:15 truenas systemd[1]: jlmkr-docker.service: Main process exited, code=exited, status=255/EXCEPTION
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ An ExecStart= process belonging to unit jlmkr-docker.service has exited.
░░ 
░░ The process' exit code is 'exited' and its exit status is 255.
Jun 22 23:13:15 truenas systemd[1]: jlmkr-docker.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The unit jlmkr-docker.service has entered the 'failed' state with result 'exit-code'.
Jun 22 23:14:35 truenas systemd[1]: Starting jlmkr-docker.service - My nspawn jail docker [created with jailmaker]...
░░ Subject: A start job for unit jlmkr-docker.service has begun execution
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit jlmkr-docker.service has begun execution.
░░ 
░░ The job identifier is 823825.
Jun 22 23:14:35 truenas .ExecStartPre[3709150]: PRE_START_HOOK
Jun 22 23:14:35 truenas systemd-nspawn[3709152]: systemd 252.22-1~deb12u1 running in system mode (+PAM +AUDIT +SELINUX +APPARMOR +IMA +SMACK +SECCOMP +GCRYPT -GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IP>
Jun 22 23:14:35 truenas systemd-nspawn[3709152]: Detected virtualization systemd-nspawn.
Jun 22 23:14:35 truenas systemd-nspawn[3709152]: Detected architecture x86-64.
Jun 22 23:14:35 truenas systemd-nspawn[3709152]: Detected first boot.
Jun 22 23:14:35 truenas systemd-nspawn[3709152]: 
Jun 22 23:14:35 truenas systemd-nspawn[3709152]: Welcome to Debian GNU/Linux 12 (bookworm)!
Jun 22 23:14:35 truenas systemd-nspawn[3709152]: 
Jun 22 23:14:35 truenas systemd-nspawn[3709152]: Initializing machine ID from container UUID.
Jun 22 23:14:35 truenas systemd-nspawn[3709152]: Failed to create control group inotify object: Too many open files
Jun 22 23:14:35 truenas systemd-nspawn[3709152]: Failed to allocate manager object: Too many open files
Jun 22 23:14:35 truenas systemd-nspawn[3709152]: [!!!!!!] Failed to allocate manager object.
Jun 22 23:14:35 truenas systemd-nspawn[3709152]: Exiting PID 1...
Jun 22 23:14:35 truenas systemd[1]: jlmkr-docker.service: Main process exited, code=exited, status=255/EXCEPTION
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ An ExecStart= process belonging to unit jlmkr-docker.service has exited.
░░ 
░░ The process' exit code is 'exited' and its exit status is 255.
Jun 22 23:14:35 truenas systemd[1]: jlmkr-docker.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ The unit jlmkr-docker.service has entered the 'failed' state with result 'exit-code'.
Jun 22 23:14:35 truenas systemd[1]: Failed to start jlmkr-docker.service - My nspawn jail docker [created with jailmaker].
░░ Subject: A start job for unit jlmkr-docker.service has failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░ 
░░ A start job for unit jlmkr-docker.service has finished with a failure.
░░ 
░░ The job identifier is 823825 and the job result is failed.`

**Comment by Jip-Hop** [comment]
https://github.com/Jip-Hop/jailmaker/issues/31

**Reply by MK00000000000000000000** [reply]
@Jip-Hop  Thank you for all your support (beyond just my questions 👍) May I ask, what are the recommended values for entering in the 'Add Sysctl' screen? I didn't see a clear response there. And I don't really know what they do. 

There's a...

> variable
> value
> description
> enabled y/n

Thanks!

**Reply by Lockszmith-GH** [reply]
There is no one-size-fits-all recommended values.

Read through the links #31 is pointing on, or just go with trial and error starting with what I've been using.

**Reply by MK00000000000000000000** [reply]
Amazing, thanks!
